crates/router/src/connector/bitpay/transformers.rs

```diff
@@ -138,7 +139,12 @@ impl<F, T>
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                // connector_response_reference_id: None,
+                connector_response_reference_id: Some(
+                    item.response
+                        .reference
+                        .unwrap_or(item.response.transaction_id),
+                ),
```

Comment:
Please revert back these changes as this issue is only about request reference ID. Remove these 6 lines.

Reply:
@Himanshu-370

Reply:
done